cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -0,0 +1,165 @@
+package workerqueue
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/astronomer/astro-cli/astro-client"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrWorkerQueueDefaultOptions = errors.New("failed to get worker-queue default options")
+	ErrInvalidWorkerQueueOption  = errors.New("worker-queue option is invalid")
+	ErrCannotUpdateExistingQueue = errors.New("worker-queue exists")
+)
+
+func Create(ws, deploymentID, name string, isDefaultWQueue bool, wQueueMin, wQueueMax, wQueueConcurrency int, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment       astro.Deployment
+		err                       error
+		errHelp                   string
+		workerQueueToCreate       *astro.WorkerQueue
+		wQueueListToCreate        []astro.WorkerQueue
+		workerQueueDefaultOptions astro.WorkerQueueDefaultOptions
+		pool                      astro.NodePool
+		nodePoolIDForWorkerQueue  string
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, client)
+	if err != nil {
+		return err
+	}
+
+	// TODO should we only get defaults once via init()
+	// get defaults for min-count, max-count and concurrency from API
+	workerQueueDefaultOptions, err = GetWorkerQueueDefaultOptions(client)
+	if err != nil {
+		return err
+	}
+
+	// Get the default nodepool ID to use
+	for _, pool = range requestedDeployment.Cluster.NodePools {
+		if pool.IsDefault {
+			nodePoolIDForWorkerQueue = pool.ID
+		}
+	}
+
+	// TODO user selects nodePoolID for creating a queue on non-default nodepools
+
+	workerQueueToCreate = &astro.WorkerQueue{
+		Name:       name,
+		IsDefault:  isDefaultWQueue,
+		NodePoolID: nodePoolIDForWorkerQueue,
+	}
+	workerQueueToCreate = setWorkerQueueValues(wQueueMin, wQueueMax, wQueueConcurrency, workerQueueToCreate, workerQueueDefaultOptions)
+
+	err = IsWorkerQueueInputValid(workerQueueToCreate, workerQueueDefaultOptions)
+	if err != nil {
+		return err
+	}
+
+	if QueueExists(requestedDeployment.WorkerQueues, workerQueueToCreate) {
+		// create does not allow updating existing queues
+		errHelp = fmt.Sprintf("use worker-queue update %s instead", workerQueueToCreate.Name)
+		return errors.Wrap(ErrCannotUpdateExistingQueue, errHelp)
+	}
+
+	// workerQueueToCreate does not exist so we add it
+	wQueueListToCreate = append(requestedDeployment.WorkerQueues, *workerQueueToCreate) //nolint
+
+	// update the deployment with the new list of worker-queues
+	err = deployment.Update(requestedDeployment.ID, "", ws, "", 0, 0, 0, wQueueListToCreate, true, client)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "worker-queue %s for %s in %s workspace created\n", workerQueueToCreate.Name, requestedDeployment.ID, ws)
+	return nil
+}
+
+// setWorkerQueueValues sets values for MinWorkerCount, MaxWorkerCount and WorkerConcurrency
+// Default values are used if the user did not request any
+
```

Comment:
```suggestion
```

Generally in golang, we don't leave an empty line between method and method's comment